src/main/java/net/rptools/maptool/client/script/javascript/api/JSAPIToken.java

```diff
@@ -199,4 +200,74 @@ public void setMap(Zone m) {
   public String getMapName() {
     return this.map.getDisplayName();
   }
+
+  @HostAccess.Export
+  public boolean getState(String stateName) {
+    Object currentState = this.token.getState(stateName);
+    return (currentState instanceof Boolean && (Boolean) currentState);
+  }
+
+  @HostAccess.Export
+  public void setState(String stateName, boolean aValue) {
+    boolean trusted = JSScriptEngine.inTrustedContext();
+    String playerId = MapTool.getPlayer().getName();
+    if (trusted || token.isOwner(playerId)) {
```

Comment:
Does this work for token that are owned by "All players" too? I think there were issues that you have to check against this separately.

Reply:
I did some testing and couldn't find an issue. When a token was set to owned by "All players" a server or client GM always succeeds at the isOwner check, and when the client is a player, not GM it succeeded as well.

I could've missed something though.

Reply:
If you tested the the case everthing is fine.